### ISSUE TITLE
Improve static assets loading and caching

### ISSUE THREAD

**Issue by Phyks** [dissemin/dissemin#687]
See for instance

https://developers.google.com/speed/pagespeed/insights/?hl=fr&url=dissem.in. 

https://developers.google.com/speed/pagespeed/insights/?hl=fr&url=https%3A%2F%2Fdissem.in%2Fsearch%2F%3Fauthors%3Dverney

https://developers.google.com/speed/pagespeed/insights/?hl=fr&url=https%3A%2F%2Fdissem.in%2Fr%2F4162%2Flucas-verney

This should have a net impact on mobile devices and search engines ranking.

**Comment by beckstefan** [comment]
For all the JS (that Google is complaining about) we can use "defer", however this would mean to make a dissemin.js (defer that too) and collect all the spread JS into that in case it relies on some JS-library that is defered.

**Comment by Phyks** [comment]
> For all the JS (that Google is complaining about) we can use "defer", however this would mean to make a dissemin.js (defer that too) and collect all the spread JS into that in case it relies on some JS-library that is defered.

Indeed, that's more or less what I was planning to do mid-term as part of #563. I'd like to investigate using a bit more modern JS stack (Yarn/NPM, Webpack) to build the JS code into single entries which could be minified and loaded more efficiently.

In this issue, I'm mainly focusing on the server side configuration so far (ensuring gzip, caching strategies and other HTTP headers).

**Comment by beckstefan** [comment]
I've done the above test on sandbox with the new bootstrap, which loads less JavaScript. There is a significant improvement (around 70 points).

However, I do not use defer, which should give us some extra points.

Another complain from Google is indeed caching. As it seems, we currently do no caching of our static files at all (nevertheless, the browser uses its cache).
